src/test/java/net/snowflake/client/jdbc/SnowflakeUtilTest.java

```diff
@@ -0,0 +1,130 @@
+    assertEquals(
+        OBJECT_MAPPER.writeValueAsString(expectedColumnMetadata),
+        OBJECT_MAPPER.writeValueAsString(columnMetadata));
+  }
+
+  private static SnowflakeColumnMetadata createExpectedMetadata(
+      JsonNode rootNode, JsonNode fieldOne, JsonNode fieldTwo) throws SnowflakeSQLLoggedException {
+    ColumnTypeInfo columnTypeInfo =
+        getSnowflakeType(rootNode.path("type").asText(), null, null, null, 0);
+    ColumnTypeInfo columnTypeInfoNodeOne =
+        getSnowflakeType(fieldOne.path("type").asText(), null, null, null, Types.BIGINT);
+    ColumnTypeInfo columnTypeInfoNodeTwo =
+        getSnowflakeType(fieldTwo.path("type").asText(), null, null, null, Types.DECIMAL);
+    SnowflakeColumnMetadata expectedColumnMetadata =
+        new SnowflakeColumnMetadata(
+            rootNode.path("name").asText(),
+            columnTypeInfo.getColumnType(),
+            rootNode.path("nullable").asBoolean(),
+            rootNode.path("length").asInt(),
+            rootNode.path("precision").asInt(),
+            rootNode.path("scale").asInt(),
+            columnTypeInfo.getExtColTypeName(),
+            false,
+            columnTypeInfo.getSnowflakeType(),
+            Arrays.asList(
+                new FieldMetadata(
+                    fieldOne.path("name").asText(),
+                    columnTypeInfoNodeOne.getExtColTypeName(),
+                    columnTypeInfoNodeOne.getColumnType(),
+                    fieldOne.path("nullable").asBoolean(),
+                    fieldOne.path("length").asInt(),
+                    fieldOne.path("precision").asInt(),
+                    fieldOne.path("scale").asInt(),
+                    fieldOne.path("fixed").asBoolean(),
+                    columnTypeInfoNodeOne.getSnowflakeType(),
+                    new ArrayList<>()),
+                new FieldMetadata(
+                    fieldTwo.path("name").asText(),
+                    columnTypeInfoNodeTwo.getExtColTypeName(),
+                    columnTypeInfoNodeTwo.getColumnType(),
+                    fieldTwo.path("nullable").asBoolean(),
+                    fieldTwo.path("length").asInt(),
+                    fieldTwo.path("precision").asInt(),
+                    fieldTwo.path("scale").asInt(),
+                    fieldTwo.path("fixed").asBoolean(),
+                    columnTypeInfoNodeTwo.getSnowflakeType(),
+                    new ArrayList<>())),
+            rootNode.path("database").asText(),
+            rootNode.path("schema").asText(),
+            rootNode.path("table").asText(),
+            false);
+    return expectedColumnMetadata;
+  }
+
+  private static ObjectNode createRootNode() {
+    ObjectNode rootNode = createFieldNode("STRUCT", 2, 128, 8, "object", false, null, 42);
+    String database = "database";
+    rootNode.put("database", database);
+    String schema = "schema";
+    rootNode.put("schema", schema);
+    String table = "table";
+    rootNode.put("table", table);
```

Comment:
1. do we need variables?
2. why do all the values have the same value as key

Reply:
Fixed